### PR TITLE
Allow piping into wallet-tx script.

### LINKED ISSUE #1621: wallet-client: Pipe to wallet-tx
Add piping mechanics to some of the wallet scripts, including `wallet-client`

### PR DESCRIPTION
**Pull request by illegalprime**
Fixes #1621

### CODE REVIEW

File: packages/bitcore-client/bin/wallet-tx [hunk 1]
```diff
@@ -18,24 +19,65 @@ program
...
+  const tx = await wallet.newTx({ addresses, amount, utxos, change, fee });
+  console.log(tx.uncheckedSerialize());
+};
+
+const checkArgs = async (args) => {
+  for (const arg of [
+    'name', 'addresses', 'amount', 'utxos', 'change', 'fee'
+  ]) {
+    if (!args[arg]) {
+      throw new Erorr(`Did not specify --${arg}`);
```
Comment (nitsujlangston) [inline packages/bitcore-client/bin/wallet-tx]: Error

File: packages/bitcore-client/bin/wallet-tx [hunk 2]
```diff
@@ -18,24 +19,65 @@ program
...
-    if (utxos) {
-      const password = await promptly.password('Wallet Password:');
-      wallet = await wallet.unlock(password);
+const loadWallet = async ({ name, path, password }) => {
+  const wallet = await Wallet.loadWallet({ name, path });
+  if (password) {
+    return await wallet.unlock(password);
+  }
+  return wallet;
+}
```
Comment (nitsujlangston) [inline packages/bitcore-client/bin/wallet-tx]: ;

File: packages/bitcore-client/bin/wallet-tx [hunk 3]
```diff
@@ -18,24 +19,65 @@ program
...
+
+  params.utxos = params.utxos ? JSON.parse(params.utxos) : [];
+  if (params.utxos) {
+    params.password = await promptly.password('Wallet Password:');
+  }
+
+  try {
+    params.addresses = JSON.parse(params.addresses);
+  }
+  catch (_) {}
```
Comment (nitsujlangston) [inline packages/bitcore-client/bin/wallet-tx]: gracefully throw

File: packages/bitcore-client/bin/wallet-tx [hunk 4]
```diff
@@ -18,24 +19,65 @@ program
...
-main();
+(async () => {
+  const { name, path, addresses, amount, utxos, change, fee } = program;
+  const params = { name, path, addresses, amount, utxos, change, fee };
+  if (process.stdin.isTTY) {
+    checkArgs(params);
+  }
+
+  params.utxos = params.utxos ? JSON.parse(params.utxos) : [];
+  if (params.utxos) {
```
Comment (micahriggan) [inline packages/bitcore-client/bin/wallet-tx]: You need to prompt for password when there aren't any utxos passed in

File: packages/bitcore-client/bin/wallet-tx [hunk 5]
```diff
@@ -18,24 +19,65 @@ program
...
+  }
+  else {
+    const rl = readline.createInterface({
+      input: process.stdin,
+      terminal: false
+    });
+    rl.on('line', async (line) => {
+      const parsed = JSON.parse(line);
+      if (parsed.utxos && !params.password) {
+        params.password = await promptly.password('Wallet Password:');
```
Comment (micahriggan) [inline packages/bitcore-client/bin/wallet-tx]: I don't think this works :( I'm trying to do a mini version of this

```
#!/usr/bin/env node

const readline = require('readline');
const promptly = require('promptly');

(async function() {
  const rl = readline.createInterface({
    input: process.stdin,
    terminal: false
  });
  let password;
  rl.on('line', async (line) => {
    console.log(line);
    password = await promptly.password('Password: ');
  })
})();
```

echo "hi" | ./test 

Doesn't prompt for password

**Comment by micahriggan** [comment]
We don't need this one anymore with the changes to the client (before you needed password to hit wallet api endpoints, but now you only need password for wallet-sign)

For wallet sign, I think we'd be piping in tx's, and we'd need to prompt for the password

**Pull request closed without merging**